bundle/src/main/java/com/adobe/acs/commons/ccvar/impl/PropertyConfigServiceImpl.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2021 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.ccvar.impl;
+
+import com.adobe.acs.commons.ccvar.PropertyConfigService;
+import com.adobe.acs.commons.ccvar.TransformAction;
+import org.apache.commons.lang3.StringUtils;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicyOption;
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.AttributeType;
+import org.osgi.service.metatype.annotations.Designate;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+@Component(service = PropertyConfigService.class)
+@Designate(ocd = PropertyConfigServiceImpl.Config.class)
+public class PropertyConfigServiceImpl implements PropertyConfigService {
+    public static final String PARSER_SEPARATOR = "|";
```

Comment:
The character | is not allowed in HTL values that represent uri's, so unfortunately tokens with actions (such as "|url") get stripped out before they can be translated.

We will need to use something other than |, such as !

`PLACEHOLDER_PATTERN` needs a corresponding update.

Reply:
Updated the separator and pattern